Tutorials/WinForms/Classic/implement-crud-functionality-with-xpo-objects.md

```diff
@@ -0,0 +1,163 @@
+* Set the `CustomerBindingSource.ObjectClassInfo` property to **DxSample.DataAccess.Customer** *(select a value from the drop-down list and rebuild the project)*.
+* Set the `CustomerBindingSource.DisplayableProperties` property to **FirstName;LastName**.
+* Set the `CustomerLayoutControl.Dock` property to **Fill**.
+* Select the `CustomerLayoutControl` component on the design surface.
+* Click the **Retrieve Fields** smart tag command to open the **Select Binding Source** wizard (if the smart tag icon is not visible, press the **Esc** key to navigate from the root `LayoutGroup` to the `LayoutControl`).
+* Change the **Data Source Update Mode** property value to **OnPropertyChanged** and click the **Next** button.
+* The **Manage Data Bindings** screen displays a list of available properties. Choose the **First Name** and **Last Name** properties, click the **Finish** button, and save changes.
+* Open the code editor and overload the constructor to accept a nullable integer value (a database record's identifier).
+    ```csharp
+    public EditCustomerForm() {
+        InitializeComponent();
+    }
+    public EditCustomerForm(int? customerId) : this() {
+        CustomerID = customerId;
+    }
+    public int? CustomerID { get; private set; }
+    ```
+* Open the `Form` designer and double click the `Form` header to add the `EditCustomerForm_Load` event handler (you can use the **Properties** window to add it). 
+* Put this code in the auto-generated `EditCustomerForm_Load` method:
+    ```csharp
+    using DevExpress.Xpo;
+    // ...
+    protected UnitOfWork UnitOfWork { get; private set; }
+    private void EditCustomerForm_Load(object sender, EventArgs e) {
+        UnitOfWork = new UnitOfWork();
+        if (CustomerID.HasValue)
+            CustomerBindingSource.DataSource = UnitOfWork.GetObjectByKey<Customer>(CustomerID.Value);
+        else CustomerBindingSource.DataSource = new Customer(UnitOfWork);
+    }
+    ```
+* Open the `CustomersListForm` designer, select the `CustomersGridView` component in the **Properties** window, and double-click the `RowClick` event in the **Events** section to create the event handler.
+* Add this code to the event handler:
+    ``` csharp
+    using DevExpress.XtraGrid.Views.Grid
+    // ...
+    private void CustomersGridView_RowClick(object sender, RowClickEventArgs e) {
+        if (e.Clicks == 2) {
+            e.Handled = true;
+            int customerID = (int)CustomersGridView.GetRowCellValue(e.RowHandle, colOid);
+            using (EditCustomerForm form = new EditCustomerForm(customerID)) {
+                form.ShowDialog(this);
+            }
+        }
+    }
+    ```
+* Press the **F5** key to run the application, double-click a `GridView` record, and see the result.
+* Close the both windows and open the `EditCustomerForm` in the designer. Drag a `SimpleButton` component from the toolbox and drop it to the `LayoutControl`. Add `EmptySpaceItem` elements to neatly align the button.
```

Comment:
"drop it to" How can you drop something 'to' something else?

Reply:
I got it! I should use "drop it on" or describe the whole process: "drag it, move it over something, and drop it". In the end, I changed this text completely using the "add" word.
ff66f17c8dcdd922e914b5836de952942340ca94